File: simulator/job.py
import itertools

class Job:
    _ids = itertools.count()

    def __init__(self, job_class, arrival_time, service_time):
        self.id = next(Job._ids)
        self.job_class = job_class
        self.arrival_time = arrival_time
        self.remaining_service = service_time

    def __repr__(self):
        return (f"Job(id={self.id}, class={self.job_class}, "
                f"arr={self.arrival_time:.4f}, rem={self.remaining_service:.4f})")

File: java/NextEventSimulator/src/main/java/com/g45/webappsim/simulator/Job.java
package com.g45.webappsim.simulator;

import java.util.concurrent.atomic.AtomicInteger;

public final class Job {
    private static final AtomicInteger SEQ = new AtomicInteger(0);

    private final int id;
    private int jobClass;               // mutable for class switch
    private final double arrivalTime;
    private double remainingService;

    public Job(int jobClass, double arrivalTime, double serviceTime) {
        this.id = SEQ.getAndIncrement();
        this.jobClass = jobClass;
        this.arrivalTime = arrivalTime;
        this.remainingService = serviceTime;
    }



    public int getId() {
        return id;
    }

    public int getJobClass() {
        return jobClass;
    }

    public void setJobClass(int c) {
        this.jobClass = c;
    }

    public double getArrivalTime() {
        return arrivalTime;
    }

    public double getRemainingService() {
        return remainingService;
    }

    public void setRemainingService(double v) {
        this.remainingService = v;
    }

    @Override
    public String toString() {
        return String.format("Job(id=%d, class=%d, arr=%.6f, rem=%.6f)", id, jobClass, arrivalTime, remainingService);
    }
    public static void reset() {
        SEQ.set(0);
    }
}
